src/process/linux.rs

```diff
@@ -0,0 +1,787 @@
+        // Wait for child process to exit.
+        unsafe {
+            let mut status: libc::c_int = 0;
+            if libc::waitpid(self.pid as i32, &mut status, 0) == -1 {
+                Err(io::Error::last_os_error())
+            } else {
+                match self.exit_signal()? {
+                    Some(exit_signal) => Ok(exit_signal),
+                    None => Ok(ExitStatus::from_raw(status)),
+                }
+            }
+        }
+    }
+
+    /// Attempts to collect the exit status of the child if it has already
+    /// exited.
+    ///
+    /// This function will not block the calling thread and will only
+    /// check to see if the child process has exited or not. If the child has
+    /// exited then on Unix the process ID is reaped. This function is
+    /// guaranteed to repeatedly return a successful exit status so long as the
+    /// child has already exited.
+    ///
+    /// If the child has exited, then `Ok(Some(status))` is returned. If the
+    /// exit status is not available at this time then `Ok(None)` is returned.
+    /// If an error occurs, then that error is returned.
+    ///
+    /// Note that unlike `wait`, this function will not attempt to drop stdin.
+    ///
+    /// # Examples
+    ///
+    /// Basic usage:
+    ///
+    /// ```no_run
+    /// use birdcage::process::Command;
+    /// use birdcage::{Birdcage, Sandbox};
+    ///
+    /// let cmd = Command::new("ls");
+    /// let mut child = Birdcage::new().spawn(cmd).unwrap();
+    ///
+    /// match child.try_wait() {
+    ///     Ok(Some(status)) => println!("exited with: {status}"),
+    ///     Ok(None) => {
+    ///         println!("status not ready yet, let's really wait");
+    ///         let res = child.wait();
+    ///         println!("result: {res:?}");
+    ///     },
+    ///     Err(e) => println!("error attempting to wait: {e}"),
+    /// }
+    /// ```
+    pub fn try_wait(&mut self) -> io::Result<Option<ExitStatus>> {
+        // Wait for child process to exit.
+        unsafe {
+            let mut status: libc::c_int = 0;
+            let pid = libc::waitpid(self.pid as i32, &mut status, libc::WNOHANG);
+            if pid == -1 {
+                Err(io::Error::last_os_error())
+            } else if pid == 0 {
+                Ok(None)
+            } else {
+                match self.exit_signal()? {
+                    Some(exit_signal) => Ok(Some(exit_signal)),
+                    None => Ok(Some(ExitStatus::from_raw(status))),
+                }
+            }
+        }
+    }
+
+    /// Simultaneously waits for the child to exit and collect all remaining
+    /// output on the stdout/stderr handles, returning an `Output`
+    /// instance.
+    ///
+    /// The stdin handle to the child process, if any, will be closed
+    /// before waiting. This helps avoid deadlock: it ensures that the
+    /// child does not block waiting for input from the parent, while
+    /// the parent waits for the child to exit.
```

Comment:
This comment makes a good point, but the code does not do what the comment says.